python/ray/data/dataset.py

```diff
@@ -3369,6 +3369,7 @@ def write_bigquery(
         self,
         project_id: str,
         dataset: str,
+        max_retry_cnt: Optional[int] = None,
```

Comment:
Rather than  `max_retry_cnt: Optional[int]` in `write_bigquery`, we could also do `max_retry_cnt: int`, and set the default here. Might make it clearer what the default behavior is.

```suggestion
        max_retry_cnt: int = 10,
```

Reply:
Sounds good